internal/datastore/common/sql.go

```diff
@@ -580,19 +702,73 @@ func (tqs QueryExecutor) ExecuteQuery(
 		limit = *queryOpts.Limit
 	}
 
-	toExecute := query.limit(limit)
+	if limit < math.MaxInt64 {
+		query = query.limit(limit)
```

Comment:
what's the impact of no longer always having a limit clause?

Reply:
As far as I can tell, nothing. It just means sending less bytes over the wire